Fix for issue with Sammelkalender_CH after update

JonasArnold opened the pull request:
These changes fix an issue which arose for some users after updating to integration version 2.11.0.

Issue:
Before version 2.11.0 the default Sammelgebiet was always used, since the API returned a single null Sammelgebiet. PR #4832 changed this. Unfortunately a new issue arose with this change. It is possible to not configure a Sammelgebiet at all (which some users did). In this case the integration now fails to fetch the waste collection dates.

Implemented Fix:
Instead of throwing an exception when the Sammelgebiet is missing (street parameter), a warning log entry is created which points out to the user that the street parameter should be configured. The first returned Sammelgebiet is then used to fetch the waste collection dates (similar behaviour as before PR #4832).

Improvements:
It would be cleaner to reconfigure all Configurations automatically to contain the default Sammelgebiet, which I do not know how to do.

5ila5 commented:
I noticed this issue as well. I thought the sensors should stay (they might get recreated).

It's how we set the unique id:
https://github.com/mampfes/hacs_waste_collection_schedule/blob/cb495700e56ddc21d1260b4d4af276f877b8d996/custom_components/waste_collection_schedule/config_flow.py#L643

So if the source arguments change, the unique id changes. But I don't really know how to fix this while allowing multiple sources for the same provider.

5ila5 commented:
I only have time for a short reply, I'll look at this in more detail soon.

This ID from Source + Municipality has two problems: I don't know how to properly do this for all sources. We don't have municipalities for all sources.: what about ICS source, where the only argument may be a URL. I don't think matching some arguments to be used to generate a unique_id and some not is a bad idea, as it does not scale to all the different sources we currently have.

Additionally, this would make it impossible to have two Hubs / Sources for the same municipalities with a different address.

markvp reviewed (changes requested):
This PR has evolved into a broader architectural discussion. Please consider splitting into separate PRs for the source-specific fix and the config_flow changes. See comment for details.

JonasArnold commented:
@markvp Thanks for picking up the topic again.
I still believe the fix I have implemented for Sammelkalender.ch is valid and can be merged. Afterall it does restore the same behaviour as before the fix I made in https://github.com/mampfes/hacs_waste_collection_schedule/pull/4832https://github.com/mampfes/hacs_waste_collection_schedule/pull/4832 but still fixes the issue for new user. Therefore it does at least work correctly for new users.

I reverted the commits back to my initial fix for sammelkalender_ch.
If my other inputs regarding the reconfig flow are of any interest, please tell me and I will create a PR with my changes and findings. But I unfortunately do not have the required time or knowledge to help with that issue.
